### ISSUE TITLE
ZODB on GCS

### ISSUE THREAD

**Issue by LoganDark** [zopefoundation/ZODB#201]
Is it possible to use ZODB with Google Cloud Storage? I can't seem to find any information anywhere about it and FileStorage only works with local files. Google App Engine does not support persisting local files.

**Comment by jimfulton** [comment]
On Sun, Apr 8, 2018 at 9:00 PM, LoganDark <notifications@github.com> wrote:

> Is it possible to use ZODB with Google Cloud Storage? I can't seem to find
> any information anywhere about it and FileStorage only works with local
> files. Google App Engine does not support persisting local files.
>
Google cloud storage looks like an alternative to S3, which wouldn't be a
backend for ZODB. OTOH, Google offers cloud versions of Postgres and MySQL
which you should be able to use with RelStorage.
http://relstorage.readthedocs.io/en/latest/

**Comment by tseaver** [comment]
Indeed, GCS sits in the same space as S3:  it uses "buckets" to store binary "blobs"/"objects".  As one of the maintainers of the `google-cloud-storage` package, I can say pretty definitely that it wouldn't be suitable for a generic ZODB replacement.  OTOH, I think it might be feasible to use GCS to handle ZODB blobs (which may be what @LoganDark was asking).

**Comment by LoganDark** [comment]
> (which may be what @LoganDark was asking)

No, my question is _exactly_ how I worded it: using ZODB entirely on Google Cloud Storage. Google App Engine does not support writing to the file system _at all_.

I'll check out relstorage.

**Comment by LoganDark** [comment]
Now there's the issue of getting relstorage to work with cloud SQL...
